inc/bans.php

```diff
@@ -113,20 +113,25 @@ static public function parse_range($mask) {
 		return array($ipstart, $ipend);
 	}
 	
-	static public function find($ip, $board = false, $get_mod_info = false) {
+	static public function find($ip, $board = false, $get_mod_info = false, $banid = false) {
 		global $config;
-		
+
+		if (!$banid)
+			$search = '(`ipstart` = :ip OR (:ip >= `ipstart` AND :ip <= `ipend`)))';
+		else
+			$search = '(``bans``.`id` = :ip))';
+
 		$query = prepare('SELECT ``bans``.*' . ($get_mod_info ? ', `username`' : '') . ' FROM ``bans``
 		' . ($get_mod_info ? 'LEFT JOIN ``mods`` ON ``mods``.`id` = `creator`' : '') . '
 		WHERE
 			(' . ($board !== false ? '(`board` IS NULL OR `board` = :board) AND' : '') . '
-			(`ipstart` = :ip OR (:ip >= `ipstart` AND :ip <= `ipend`)))
+		' . $search . '
```

Comment:
and this is the main reason this is not smart or good code. we should eliminate as many direct string manipulation of queries as possible.